packages/cozy-client/src/models/file.js

```diff
@@ -622,6 +622,68 @@ export const fetchBlobFileById = async (client, fileId) => {
   return fileBlob
 }
 
+/**
+ * @param {IOCozyFile} file - io.cozy.files document
+ * @returns {boolean}
+ */
+const isExpiringFrenchNationalIdCard = file => {
+  const label = get(file, 'metadata.qualification.label', null)
+  const country = get(file, 'metadata.qualification.country', null)
+  const expirationDate = get(file, 'metadata.expirationDate', null)
+  if (
+    label === 'national_id_card' &&
+    (!country || country === 'fr') &&
+    expirationDate
+  ) {
+    return true
+  }
+  return false
+}
+
+/**
+ * @param {IOCozyFile} file - io.cozy.files document
+ * @returns {boolean}
+ */
+const isExpiringResidencePermit = file => {
+  const label = get(file, 'metadata.qualification.label', null)
+  const expirationDate = get(file, 'metadata.expirationDate', null)
+  if (label === 'residence_permit' && expirationDate) {
+    return true
+  }
+  return false
+}
+
+/**
+ * @param {IOCozyFile} file - io.cozy.files document
+ * @returns {boolean}
+ */
+const isExpiringPersonalSportingLicense = file => {
+  const label = get(file, 'metadata.qualification.label', null)
+  const referencedDate = get(file, 'metadata.referencedDate', null)
+  const created_at = get(file, 'created_at', null)
+  if (label === 'personal_sporting_licence' && (referencedDate || created_at)) {
+    return true
+  }
+  return false
+}
```

Comment:
We should start coding differently, something more declarative instead of having those imperative `if`... We should only rely on a configuration file and the code will do the rest: 


```
const config = [
'frenchNationalCard: {
'isExpiring': true, 
'expiringCondition': file => .... 
}
]
```

Like that there is no need to update the code if we want to add a new paper, change expiring condition or else. We just have to edit this config. If we think even further this config can be a json file served directly by the cozy. Like that even the PO will be able to edit by him/herself the content.

Reply:
The original approach was indeed to declare everything in a JSON (`paperDefinitions` in the first commit, that is later removed), but we discussed extensively between devs and decided to move away from a declarative coding since the conditions to determine if a given paper expires and how to compute their expiration and notice dates are already very different. 

For each relevant paper, the expiring condition is different because we use different attributes and not as many. Moreover the way we compute dates is also already different, with different fallback behavior and default delays. Making something purely declarative and thus generic would need many parameters and add noise in my opinion. As you first suggested it is easier to define dedicated methods for each paper anyway, so at this point i don't really see the benefit to define either a very complex JSON config that won't really be extensible in practice or methods instead of functions.

Note that we also discussed this with the PO and confirmed that were was no need for them to be able to edit themselves a config.

Reply:
> we discussed extensively between devs and decided to move away from a declarative coding

Attention à ne pas mélanger tous les sujets. A mon sens lors de nos échanges nous n'avons pas exclu spécifiquement le fait du faire du déclaratif de manière absolu. Là dessus, par défaut, perso je suis pas fermé.

Maintenant le fait que ce soit facile ou pas, et pertinent ou non (ce que tu évoques) est un autre point.